packages/block-library/src/image/edit.js

```diff
@@ -304,15 +310,7 @@ export function ImageEdit( {
 					onSelectImage( img );
 				},
 				allowedTypes: ALLOWED_MEDIA_TYPES,
-				onError: ( message ) => {
-					isTemp = false;
```

Comment:
@glendaviesnz I reverted the change from #34371. Settings `temporaryURL` to undefined also fixed the issue.